benchmarks/newton_solver_benchmark_set/spiegelman_et_al_2016/drucker_prager_compositions.cc

```diff
@@ -0,0 +1,675 @@
+
+        std::vector<double> cohesion;
+        std::vector<double> phi;
+
+        std::vector<double> prefactor;
+        std::vector<double> stress_exponent;
+
+        unsigned int n_fields;
+        // averaging parameters
+        double viscosity_averaging_p;
+
+        bool use_analytical_derivative;
+
+        //optimize variables
+        const double sqrt_3 = std::sqrt(3);
+        const double sqrt_half = std::sqrt(0.5);
+        std::vector<double> sin_phi;
+        std::vector<double> cos_phi;
+
+
+
+    };
+
+  }
+}
+
+#endif
+
+#include <aspect/utilities.h>
+#include <aspect/parameters.h>
+
+using namespace dealii;
+
+namespace aspect
+{
+  namespace
+  {
+    std::vector<double>
+    get_vector_double (const std::string &parameter, const unsigned int n_fields, ParameterHandler &prm)
+    {
+      std::vector<double> parameter_list;
+      parameter_list = Utilities::string_to_double(Utilities::split_string_list(prm.get (parameter)));
+      if (parameter_list.size() == 1)
+        parameter_list.resize(n_fields, parameter_list[0]);
+
+      AssertThrow(parameter_list.size() == n_fields,
+                  ExcMessage("Length of " + parameter + " list (size "+ std::to_string(parameter_list.size()) +") must be either one,"
+                             " or n_compositional_fields+1 (= " + std::to_string(n_fields) + ")."));
+
+      return parameter_list;
+    }
+  }
+
+  namespace MaterialModel
+  {
+
+    template <int dim>
+    double
+    DruckerPragerCompositions<dim>::
+    compute_second_invariant(const SymmetricTensor<2,dim> strain_rate, const double min_strain_rate) const
+    {
+      const double edot_ii_strict = std::sqrt(strain_rate*strain_rate);
+      const double edot_ii = std::max(edot_ii_strict, min_strain_rate*min_strain_rate);
+      return edot_ii;
+    }
+
+    template <int dim>
+    double
+    DruckerPragerCompositions<dim>::
+    compute_viscosity(const double edot_ii,const double pressure,const int comp,const double prefactor,const bool regularize, const double min_visc, const double max_visc) const
+    {
+      double viscosity;
+      if (comp == 0)
+        {
+          const double strength = ( (dim==3)
+                                    ?
+                                    ( 6.0 * cohesion[comp] * cos_phi[comp] + 6.0 * std::max(pressure,0.0) * sin_phi[comp] )
+                                    / ( sqrt_3 * ( 3.0 + sin_phi[comp] ) )
+                                    :
+                                    cohesion[comp] * cos_phi[comp] + std::max(pressure,0.0) * sin_phi[comp] );
+
+          // Rescale the viscosity back onto the yield surface
+          if (strength != 0 && edot_ii != 0)
+            viscosity = strength / ( 2.0 * sqrt_half * edot_ii );
+          else
+            viscosity = ref_visc;
+          if (regularize == true)
+            viscosity = ref_visc*viscosity / (ref_visc + viscosity);
+        }
+      else
+        {
+          viscosity = prefactor;
```

Comment:
```suggestion
          viscosity = ref_visc;
```
Would the above work? This would allow completely removing variable `prefactor` and all related code.

Reply:
The prefactor for composition != 0 is 1e21, while the reference viscosity is 5e24, so no, that would not be the same.